server/src/server.js

```diff
@@ -675,10 +696,23 @@ exports.simpleResponse = simpleResponse;
 exports.errorResponse = errorResponse;
 
 linker.init().then(() => {
-  app.listen(config.port);
-  console.info(`server listening on http://localhost:${config.port}/`);
-  contentApp.listen(config.contentPort);
-  console.info(`content server listening on http://localhost:${config.contentPort}/`);
+  if (config.useVirtualHosts) {
+    const mainapp = express();
+    const siteName = config.siteOrigin.split(":")[0];
+    const contentName = config.contentOrigin.split(":")[0];
+    mainapp.use(vhost(siteName, app));
+    mainapp.use(vhost(contentName, contentApp));
+    mainapp.listen(config.port);
+    mainapp.get("/", function (req, res) {
+      res.send("ok");
+    });
+    console.info(`virtual host server listening on http://localhost:${config.port}`);
+  } else {
+    app.listen(config.port);
+    console.info(`server listening on http://localhost:${config.port}/`);
+    contentApp.listen(config.contentPort);
+    console.info(`content server listening on http://localhost:${config.contentPort}/`);
+  }
```

Comment:
(The docs on vhost aren't explicit about port matching, so I'm not actually sure if that works)